triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala

```diff
@@ -70,6 +72,7 @@ class Server(dar: Option[Dar[(PackageId, Package)]], jdbcConfig: Option[JdbcConf
 
   private var triggers: Map[UUID, RunningTrigger] = Map.empty;
   private var triggersByToken: Map[Jwt, Set[UUID]] = Map.empty;
+  private var triggerLog: Map[UUID, List[(String, String)]] = Map.empty;
```

Comment:
Two things here:
1. The `(String, String)` is quite mysterious. I'd either define a data type with a `DateTime` field (there's probably an existing one) or just make it a single `String` as it is a log message.
2. Any reason you're using a (linked) `List` here? Not an expert on Scala collections but I think `Vector` would be more standard. Then you would append to the log rather than keeping it in reverse order.

Reply:
Good catch on the vector thing. Switching to that. Re the pair of strings thing, I think it's ok. Let's see how it goes - can always generalize particularly if we end up stashing more here.